[OP#41900] Improve developers documentation (#132)
nextcloud/integration_openproject: README.md

# OpenProject integration into Nextcloud

> OpenProject integration provides a dashboard widget for displaying notifications on changes in active work packages,
> a search provider for work packages, through which you can search, link, and unlink work packages to a particular file.

Reviewer: ```suggestion
OpenProject integration provides a dashboard widget for displaying notifications of changes in active work packages,
and a search provider for work packages, through which you can search, link, and unlink work packages to a particular file.
```

> OpenProject integration provides a dashboard widget for displaying notifications of changes in active work packages, and
> a search provider for work packages, through which you can search, link, and unlink work packages to a particular file.

Reviewer 2: I would put the linking of WP first, because that became our most important feature
I think the search provider means the searching through the global search of NC
the dashboard does not check for changes in WP, but reads the OP notifications
also we can display the count of OP notifications in NC notification
```suggestion
OpenProject integration provides the possibility to link OpenProject work-packages to a file in NextCloud, a dashboard widget for displaying OpenProject notifications, a search provider to search for work packages through the unified search and can display the count of OpenProject notifications as NextCloud notification.
```

### Start compose
**Note:** If your host machine has anything up on port `80`, please kill it before starting. 

It is highly recommended to regularly update the included containers.
```shell
docker-compose pull
```

Now, run the containers.
```shell
docker-compose up
```

> **Note:** If you've cloned integration app anywhere other that the default `./../../custom_apps`, provide its path in `APP_DIR` environment variable

Reviewer: ```suggestion
**Note:** If you've cloned the integration app anywhere other than the default `./../../custom_apps`, provide its path in the `APP_DIR` environment variable
```

```shell
APP_DIR = <path-to-integration-app> docker-compose up
```

> After this, you should be able to access the nextcloud server at [http://localhost](http://localhost).

Reviewer 3: ```suggestion
After this, you should be able to access the Nextcloud server at [http://localhost](http://localhost).
```

cd $HOME/development/custom_apps/integration_openproject
npm run watch
```

> > **Note:** Inorder to use the search bar for searching work packages you need to set-up file storages in openProject side for your particular nextcloud instance. To do so you can follow this [documentation](https://google.com)

Reviewer 3: ```suggestion
> **Note:** In order to use the search bar for searching work packages you need to set up file storage on the OpenProject side for your particular Nextcloud instance. To do so you can follow this [documentation](https://google.com)
```

> > **Note:** Inorder to use the search bar for searching work packages you need to set-up file storages in openProject side for your particular nextcloud instance. To do so you can follow this [documentation](https://google.com)

Reviewer: ```suggestion
> **Note:** In order to use the search bar for searching work packages you need to set-up file storages in the openProject side for your particular nextcloud instance. To do so you can follow this [documentation](https://google.com)
```

Author: Yes for now there's no documentation for how to set it up from the OpenProject side, we will have to link it once there's a proper docs

> > **Note:** In order to use the search bar for searching work packages you need to set up file storages on the OpenProject side for your particular Nextcloud instance. To do so you can follow this [documentation](https://www.openproject.org/docs/)

Reviewer 2: maybe that could go to the top, because it will be needed also if people want just to run the app and not develop it

cd $HOME/development/custom_apps/integration_openproject
npm run watch
```

> ### Setting up storages

Reviewer 2: I believe this documentation should be all in OpenProject, here we only want to have documentation about how to start development of the nextcloud app
